packages/datadog-instrumentations/src/apollo-server-core.js

```diff
@@ -1,36 +1,38 @@
 'use strict'
 
 const { AbortController } = require('node-abort-controller')
-const { addHook, channel } = require('./helpers/instrument')
+const { addHook } = require('./helpers/instrument')
 const shimmer = require('../../datadog-shimmer')
+const dc = require('dc-polyfill')
 
-const startApolloServerCoreRequest = channel('datadog:apollo-server-core:request:start')
-const successApolloServerCoreRequest = channel('datadog:apollo-server-core:request:success')
+const requestChannel = dc.tracingChannel('datadog:apollo-server-core:request')
 
 addHook({ name: 'apollo-server-core', file: 'dist/runHttpQuery.js', versions: ['>3.0.0'] }, runHttpQueryModule => {
   const HttpQueryError = runHttpQueryModule.HttpQueryError
 
   shimmer.wrap(runHttpQueryModule, 'runHttpQuery', function wrapRunHttpQuery (originalRunHttpQuery) {
     return async function runHttpQuery () {
-      if (!startApolloServerCoreRequest.hasSubscribers) {
+      if (!requestChannel.start.hasSubscribers) {
         return originalRunHttpQuery.apply(this, arguments)
       }
 
-      startApolloServerCoreRequest.publish()
-
-      const runHttpQueryResult = await originalRunHttpQuery.apply(this, arguments)
-
       const abortController = new AbortController()
       const abortData = {}
-      successApolloServerCoreRequest.publish({ abortController, abortData })
-
-      if (abortController.signal.aborted) {
-        // runHttpQuery callbacks are writing the response on resolve/reject.
-        // We should return blocking data in the apollo-server-core HttpQueryError object
-        return Promise.reject(new HttpQueryError(abortData.statusCode, abortData.message, true, abortData.headers))
-      }
 
-      return runHttpQueryResult
+      const runHttpQueryResult = requestChannel.tracePromise(
+        originalRunHttpQuery,
+        { abortController, abortData },
+        this,
+        ...arguments)
+
+      return Promise.race([runHttpQueryResult]).then((value) => {
+        if (abortController.signal.aborted) {
+          // runHttpQuery callbacks are writing the response on resolve/reject.
+          // We should return blocking data in the apollo-server-core HttpQueryError object
+          return Promise.reject(new HttpQueryError(abortData.statusCode, abortData.message, true, abortData.headers))
+        }
+        return value
+      })
```

Comment:
The `Promise.race(...)` is unnecessary unless you're actually going to do a race with the abort signal, but here you're just handling it in the continuation. If you want to make proper use of the race you'd have to do something like this:

```suggestion
      const abortPromise = new Promise((resolve, reject) => {
        abortController.signal.addEventListener('abort', (event) => {
          // runHttpQuery callbacks are writing the response on resolve/reject.
          // We should return blocking data in the apollo-server-core HttpQueryError object
          reject(new HttpQueryError(abortData.statusCode, abortData.message, true, abortData.headers))
        }, { once: true })
      })

      return Promise.race([
        runHttpQueryResult,
        abortPromise
      ])
```

Reply:
Yeah sorry, I misinterpreted your previous comment. I've already addressed it. Thanks.